### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by chalcolith**
This PR adds a Windows `make.bat` script and modifies some of the shell commands to support Windows.

### CODE REVIEW

**Comment by chalcolith** [comment]
Hmm, the syntax for setting PONYPATH is slightly broken.  Hold off until I fix it, please.

**Comment by chalcolith** [comment]
Should be good to go now.

File: stable/main.pony [hunk 1]
```diff
@@ -63,9 +63,22 @@ actor Main
...
-        ["env"; "PONYPATH="+ponypath].>append(rest), env~exitcode()
-      )?
+      ifdef windows then
+        var cmd: String trn = recover String end
+        cmd.append("cmd /C \"set \"PONYPATH=")
+        cmd.append(ponypath)
+        cmd.append("\" &&")
+        for arg in rest.values() do
+          cmd.append(" ")
+          cmd.append(arg)
```
Comment (jemc) [inline stable/main.pony:73]: Does windows need any escaping here if the argument contains special characters, like a space?

Comment (chalcolith) [reply stable/main.pony:73]: The command ends up being `set "PONYPATH=foo"`; those quotes will take care of any spaces.

Comment (chalcolith) [reply stable/main.pony:73]: And they don't need to be escaped in the context of the outer quotes.  Windows is weird.

Comment (jemc) [reply stable/main.pony:73]: Definitely sounds, weird, but I'll trust you! :wink:

**Pull request merged**